sqlx-test/src/lib.rs

```diff
@@ -223,3 +230,10 @@ macro_rules! Postgres_query_for_test_prepared_type {
         "SELECT ({0} is not distinct from $1)::int4, {0}, $2"
     };
 }
+
+#[macro_export]
+macro_rules! Postgres_query_for_test_prepared_geometric_type {
+    () => {
+        "SELECT ({0}::text is not distinct from $1::text)::int4, {0}, $2"
```

Comment:
Why is this necessary? What happens without it?

Reply:
`could not identify an equality operator for type point`

We can't use the `SELECT ({0} is not distinct from $1)::int4, {0}, $2` from https://github.com/jayy-lmao/sqlx/blob/feat/geometry-postgres-point/sqlx-test/src/lib.rs#L230 as the comparison

Reply:
I see. `=` compares _area_ for the two-dimensional types, and there's a different operator for equality, `~=`.

https://www.postgresql.org/docs/current/functions-geometry.html

We should use that rather than converting from text because there could be differences in rounding between Postgres and Rust that alter the text output of the types.

Reply:
Took a bit of a fiddle, and needed to split between array/non-array variants. 

This is what worked in the end: https://github.com/launchbadge/sqlx/pull/3583/files#diff-31a2fb7770cf7cd19a7baadb82c0ccfa1721d92c1eeb4d3f7f9a675de54f20b0R247-R251